collector/logical_router_collector.go

```diff
@@ -0,0 +1,90 @@
+package collector
+
+import (
+	"nsxt_exporter/client"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	nsxt "github.com/vmware/go-vmware-nsxt"
+	"github.com/vmware/go-vmware-nsxt/manager"
+)
+
+func init() {
+	registerCollector("logical_router", createLogicalRouterCollectorFactory)
+}
+
+type logicalRouterCollector struct {
+	logicalRouterClient client.LogicalRouterClient
+	logger              log.Logger
+
+	natTotalPackets *prometheus.Desc
+	natTotalBytes   *prometheus.Desc
+}
+
+type logicalRouterNatStatisticMetric struct {
+	LogicalRouterID string
+	NatTotalPackets float64
+	NatTotalBytes   float64
+}
+
+func createLogicalRouterCollectorFactory(apiClient *nsxt.APIClient, logger log.Logger) prometheus.Collector {
+	nsxtClient := client.NewNSXTClient(apiClient, logger)
+	return newLogicalRouterCollector(nsxtClient, logger)
+}
+
+func newLogicalRouterCollector(logicalRouterClient client.LogicalRouterClient, logger log.Logger) *logicalRouterCollector {
+	natTotalPackets := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "logical_router", "nat_total_packets"),
+		"Total packets processed by the NAT rule associated with logical router",
+		[]string{"logical_router_id"},
+		nil,
+	)
+	natTotalBytes := prometheus.NewDesc(
```

Comment:
These metrics are for all nat rules in a single router. Are we going to have the statistic for individual nat rules?